### PR TITLE
Replace chosen with select2 - rebase

### PR DESCRIPTION
**Pull request by johnnyshields**
Rebase of #327 to latest master

### CODE REVIEW

**Comment by johnnyshields** [comment]
Please merge

**Comment by CloCkWeRX** [comment]
```
should create a contact
     Failure/Error: scope :my, ->(current_user) { accessible_by(current_user.ability) }
     
     ActionView::Template::Error:
       wrong number of arguments (given 0, expected 1)
     # ./lib/fat_free_crm/permissions.rb:25:in `block in uses_user_permissions'
     # ./app/helpers/accounts_helper.rb:31:in `account_select'
     # ./app/helpers/accounts_helper.rb:57:in `account_select_or_create'
     # ./app/views/contacts/_top_section.html.haml:29:in `block (2 levels) in _app_views_contacts__top_section_html_haml__1283963717413814871_143958460'
     # ./app/views/contacts/_top_section.html.haml:23:in `block in _app_views_contacts__top_section_html_haml__1283963717413814871_143958460'
     # ./lib/fat_free_crm/callback.rb:125:in `hook'
     # ./app/views/contacts/_top_section.html.haml:2:in `_app_views_contacts__top_section_html_haml__1283963717413814871_143958460'
     # ./app/views/contacts/_new.html.haml:8:in `block in _app_views_contacts__new_html_haml___3522022147577464351_123488920'
     # ./lib/fat_free_crm/gem_ext/simple_form/action_view_extensions/form_helper.rb:21:in `block in simple_form_for'
     # ./lib/fat_free_crm/gem_ext/simple_form/action_view_extensions/form_helper.rb:20:in `simple_form_for'
     # ./app/views/contacts/_new.html.haml:1:in `_app_views_contacts__new_html_haml___3522022147577464351_123488920'
     # ./app/views/contacts/new.js.haml:8:in `_app_views_contacts_new_js_haml___1176006281886579846_133134020'
     # ./app/controllers/entities/contacts_controller.rb:47:in `new'
     # ------------------
     # --- Caused by: ---
     # ArgumentError:
     #   wrong number of arguments (given 0, expected 1)
     #   ./lib/fat_free_crm/permissions.rb:25:in `block in uses_user_permissions'
```

File: app/helpers/accounts_helper.rb [hunk 1]
```diff
@@ -27,13 +27,11 @@ def account_summary(account)
...
   def account_select(options = {})
-    options[:selected] = (@account&.id) || 0
-    accounts = ([@account] + Account.my(current_user).order(:name).limit(25)).compact.uniq
-    collection_select :account, :id, accounts, :id, :name, options,
-                      "data-placeholder": t(:select_an_account),
-                      "data-url": auto_complete_accounts_path(format: 'json'),
-                      style: "width:330px; display:none;",
-                      class: 'ajax_chosen'
+    options[:selected] = @account&.id || 0
+    accounts = ([@account.new_record? ? nil : @account] + Account.my.order(:name).limit(25)).compact.uniq
```
Comment (CloCkWeRX) [inline app/helpers/accounts_helper.rb]: .my(current_user) missed?

File: app/assets/javascripts/crm.js.coffee [hunk 2]
```diff
@@ -105,20 +104,19 @@
...
     select_account: ->
-      crm.makeAjaxChosen()
       $("#account_disabled_title").hide()
       $("#account_create_title").hide()
       $("#account_select_title").show()
       $("#account_name").hide()
       $("#account_name").prop('disabled', true)
       $("#account_id").prop('disabled', false)
-      $("#account_id_chzn").show()
+      $("#account_id").show()
```
Comment (johnnyshields) [inline app/assets/javascripts/crm.js.coffee]: not sure if this line is needed

**Comment by johnnyshields** [comment]
This PR needs to be tested manually

**Comment by CloCkWeRX** [comment]
Let's use it as an opportunity to see what feature test coverage is missing too; because in theory most of these views should be covered

**Comment by johnnyshields** [comment]
@mesmerze can you help with this?

**Comment by johnnyshields** [comment]
Replaced by a different PR.

**Pull request closed without merging**